### PR TITLE
Bug/full stops in chart images titles

### PR DESCRIPTION
**Pull request by daiLlew**
Fix for chart image downloads where chart there is period in the title.

### CODE REVIEW

File: src/main/java/com/github/onsdigital/babbage/highcharts/ChartRenderer.java [hunk 1]
```diff
@@ -39,18 +63,14 @@ private ChartRenderer() {
...
-            Integer width = getWidth(request);
-            if (width == null) {
-                width = DEFAULT_CHART_WIDTH;
-            }
-            additionalData.put("width", width);
-            String chartConfig = TemplateService.getInstance().renderChartConfiguration(contentResponse.getDataStream(),
+            ContentResponse contentResponse = contentClient.getContent(uri);
+            Map<String, Object> additionalData = new ChartConfigBuilder().width(getWidth(request)).getMap();
+
+            String chartConfig = templateService.renderChartConfiguration(contentResponse.getDataStream(),
```
Comment (daiLlew) [inline src/main/java/com/github/onsdigital/babbage/highcharts/ChartRenderer.java]: good spot. Given the grief open streams have caused us recently I am slightly embarrassed I missed this.

**Comment by Crispioso** [comment]
From a user perspective the functionality all seems to work as expected now, so it's got my 👍

File: src/main/java/com/github/onsdigital/babbage/api/endpoint/chart/ChartImage.java [hunk 2]
```diff
@@ -18,6 +19,10 @@
 
     @GET
     public void get(@Context HttpServletRequest request, @Context HttpServletResponse response) throws IOException, ContentReadException {
-        ChartRenderer.getInstance().renderChartImage(request, response);
+        try {
+            ChartRenderer.getInstance().renderChartImage(request, response);
+        } catch (Exception ex) {
```
Comment (daiLlew) [inline src/main/java/com/github/onsdigital/babbage/api/endpoint/chart/ChartImage.java]: done.

File: src/main/java/com/github/onsdigital/babbage/highcharts/ChartRenderer.java [hunk 3]
```diff
@@ -60,82 +80,63 @@ public void renderChartConfig(HttpServletRequest request, HttpServletResponse re
...
+            Map<String, Object> additionData = new ChartConfigBuilder()
+                    .width(getWidth(request))
+                    .showTitle(request)
+                    .showSubTitle(request)
+                    .showSource(request)
+                    .showNotes(request)
+                    .getMap();
+
+            String renderedTemplate = templateService.renderTemplate(EMBEDED_HIGHCHARTS_TEMPLATE,
+                    contentResponse.getDataStream(), additionData);
```
Comment (daiLlew) [inline src/main/java/com/github/onsdigital/babbage/highcharts/ChartRenderer.java]: done.

File: src/main/java/com/github/onsdigital/babbage/highcharts/ChartRenderer.java [hunk 4]
```diff
@@ -60,82 +80,63 @@ public void renderChartConfig(HttpServletRequest request, HttpServletResponse re
...
-            LinkedHashMap<String, Object> additionalData = new LinkedHashMap<>();
-            additionalData.put("width", width);
-            String chartConfig = TemplateService.getInstance().renderChartConfiguration(contentResponse.getDataStream(),
-                    additionalData);
-            InputStream stream = HighChartsExportClient.getInstance().getImage(chartConfig, getWidth(request));
-            new BabbageContentBasedBinaryResponse(contentResponse,stream, "image/png").apply(request, response);
+            Map<String, Object> additionalData = new ChartConfigBuilder().width(width).getMap();
+
+            String chartConfig = templateService.renderChartConfiguration(contentResponse.getDataStream(), additionalData);
+            InputStream stream = highChartsExportClient.getImage(chartConfig, width);
```
Comment (daiLlew) [inline src/main/java/com/github/onsdigital/babbage/highcharts/ChartRenderer.java]: done

File: src/main/java/com/github/onsdigital/babbage/highcharts/ChartRenderer.java [hunk 5]
```diff
@@ -144,13 +145,15 @@ public void renderChartConfigFor(HttpServletRequest request, HttpServletResponse
...
         if (assertUri(uri, request, response)) {
-            ContentResponse contentResponse = ContentClient.getInstance().getContent(uri);
-            LinkedHashMap<String, Object> additionalData = new LinkedHashMap<>();
-            additionalData.put("width", getWidth(request));
-            new BabbageContentBasedStringResponse(contentResponse, TemplateService.getInstance().renderTemplate("highcharts/chart", contentResponse.getDataStream(), additionalData),
-                    MediaType.TEXT_HTML).apply(request, response);
+            ContentResponse contentResponse = contentClient.getContent(uri);
+            Map<String, Object> additionalData = new ChartConfigBuilder().width(getWidth(request)).getMap();
+            String renderedTemplate = templateService.renderTemplate(HIGHCHARTS_TEMPLATE,
+                    contentResponse.getDataStream(), additionalData);
```
Comment (daiLlew) [inline src/main/java/com/github/onsdigital/babbage/highcharts/ChartRenderer.java]: done.

File: src/main/java/com/github/onsdigital/babbage/highcharts/HighChartsExportClient.java [hunk 6]
```diff
@@ -61,9 +62,16 @@ public InputStream getImage(String chartConfig, Integer width, Double scale) thr
             postParameters.add(new BasicNameValuePair("scale", scale.toString()));
         }
         //postParameters.add(new BasicNameValuePair("async", "false"));
-        CloseableHttpResponse response = client.sendPost("/", null, postParameters);
-        System.out.println("Highcharts export response: " + response.getStatusLine());
-        return response.getEntity().getContent();
+        try {
+            CloseableHttpResponse response = client.sendPost("/", null, postParameters);
```
Comment (daiLlew) [inline src/main/java/com/github/onsdigital/babbage/highcharts/HighChartsExportClient.java]: done.

**Pull request merged**